Skip test_drift_detector if required test file is missing

bdrung opened the pull request:
The release tarball does not contain the file
`scripts/suse/zypper/plugins/commit/zyppnotify`, which causes
`ZyppPluginsTestCase.test_drift_detector` to fail:

```
ERROR: test_drift_detector (unit.test_zypp_plugins.ZyppPluginsTestCase)
[CPU:0.0%|MEM:5.5%]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/unit/test_zypp_plugins.py", line 45, in test_drift_detector
    zyppnotify = imp.load_source('zyppnotify', ZYPPNOTIFY_FILE)
  File "/usr/lib/python3.7/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 724, in exec_module
  File "<frozen importlib._bootstrap_external>", line 859, in get_code
  File "/usr/lib/python3.7/imp.py", line 157, in get_data
    return super().get_data(path)
  File "<frozen importlib._bootstrap_external>", line 916, in get_data
FileNotFoundError: [Errno 2] No such file or directory: 'scripts/suse/zypper/plugins/commit/zyppnotify'
```

Therefore skip `ZyppPluginsTestCase.test_drift_detector` if `zyppnotify`
cannot be found.

bdrung committed:
Skip test_drift_detector if required test file is missing

The release tarball does not contain the file
`scripts/suse/zypper/plugins/commit/zyppnotify`, which causes
`ZyppPluginsTestCase.test_drift_detector` to fail:

```
ERROR: test_drift_detector (unit.test_zypp_plugins.ZyppPluginsTestCase)
[CPU:0.0%|MEM:5.5%]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/unit/test_zypp_plugins.py", line 45, in test_drift_detector
    zyppnotify = imp.load_source('zyppnotify', ZYPPNOTIFY_FILE)
  File "/usr/lib/python3.7/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 724, in exec_module
  File "<frozen importlib._bootstrap_external>", line 859, in get_code
  File "/usr/lib/python3.7/imp.py", line 157, in get_data
    return super().get_data(path)
  File "<frozen importlib._bootstrap_external>", line 916, in get_data
FileNotFoundError: [Errno 2] No such file or directory: 'scripts/suse/zypper/plugins/commit/zyppnotify'
```

Therefore skip `ZyppPluginsTestCase.test_drift_detector` if `zyppnotify`
cannot be found.

Signed-off-by: Benjamin Drung <benjamin.drung@cloud.ionos.com>

DmitryKuzmenko commented:
@bdrung thank you for PR.
@dmurphy18 I'm just not sure isn't it better to include the `scripts/suse/zypper/plugins/commit/zyppnotify` file into the package to not disable the test?

dmurphy18 reviewed (commented):
Changes look good.  as for 
```
I'm just not sure isn't it better to include the scripts/suse/zypper/plugins/commit/zyppnotify file into the package to not disable the test?
```

SUSE handle the packaging for SLES, and yest it would be better for them to include the file.
@brejoc perhaps you should include the file in the packaging so test do not get skipped.

bdrung commented:
IMO we should do both. Skip the test if the file is missing and ensure that this file is included in the release tarball.

bdrung committed:
test_zypp_plugins: Protect import zypp_plugin by try/except

pylint complains:

tests/unit/test_zypp_plugins.py:17: [W8410(3rd-party-module-not-gated),
] 3rd-party module import is not gated in a try/except: 'zypp_plugin'

Signed-off-by: Benjamin Drung <benjamin.drung@cloud.ionos.com>

dmurphy18 committed:
Merge branch 'master' into master-skip-test_drift_detector-if-zyppnotify-is-missing

dmurphy18 committed:
Merge branch 'master' into master-skip-test_drift_detector-if-zyppnotify-is-missing
